// prolob/lib/dataprovider.js
var mysql      = require('mysql');
var connection = mysql.createConnection({
  host     : 'localhost',
  database : 'nodeerp',
  user     : 'root',
  password : 'root'
});



function Dataprovider () {



}

Dataprovider.prototype.connect = function(){

   connection.connect();

   connection.query('SELECT 1 + 1 AS solution', function(err, rows, fields) {
     if (err) throw err;

     console.log('The solution is: ', rows[0].solution);
   });

   connection.end();
    //this.Connection.connect();

    //this.Connection.query('SELECT 1 + 1 AS solution', function(err, rows, fields) {
     // if (err) throw err;

       // console.log('The solution is: ', rows[0].solution);
   // });

    //this.Connection.end();

    return 'connection open';
};


Dataprovider.prototype.insert = function(tablename,values){
    "use strict";

    var sql =  'INSERT INTO ' + tablename +  ' SET ?';

   connection.connect();
   connection.query(sql,values,function(err,result){
        console.log(err);
   });

   connection.end();

};

Dataprovider.prototype.update = function(tablename,values){
    "use strict";

    var sql = ''

};





module.exports = exports =  Dataprovider;
